src/main/java/ch/njol/skript/doc/Experimental.java

```diff
@@ -0,0 +1,23 @@
+package ch.njol.skript.doc;
+
+import org.skriptlang.skript.lang.experiment.LifeCycle;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Provides information about an experiment which the annotated element requires.
+ */
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+public @interface Experimental {
+
+	String featureName();
+	LifeCycle phase();
```

Comment:
Do you think it might be better to require the actual experiment here? It seems tedious to have to update the phase whenever an experiment updates. It can just be grabbed from the enum: https://github.com/SkriptLang/Skript/blob/ef3d3603c4aaedf519f2c1322c3a0f09f4b53a67/src/main/java/ch/njol/skript/registrations/Feature.java

Reply:
I was thinking this as well. Maybe it would be better to have the annotation be like, `@Feature(String featureName, String feedbackLink)`, and then use smth like `ExperimentRegistry#find(String name)` later to get the experiment and its phase and stuff.

Reply:
On our side it would definitely be easier to just use the Feature enum itself in the annotation, but obviously that means an addon couldn't use the same thing for its own documentation.

There is a bit of trickery you can do with annotations inside annotations, I'll see if I can find a solution that supports both.